### PR TITLE
statistics: batch insert topn and buckets to storage to speed up loading stats

### LINKED ISSUE #36947: speed up load stats procedure
## Enhancement

When loading stats from json file, each of TopN(Buckets) is inserted into `mysql.stats_top_n`(`mysql.stats_buckets`) by one internal insert SQL, which is inefficient. We can batch insert  TopN(Buckets) to speed up load stats procedure.

### PR DESCRIPTION
**Pull request by xuyifangreeneyes**
### What problem does this PR solve?

Issue Number: close #36947

Problem Summary:

### What is changed and how it works?

Batch insert topn and buckets to storage to speed up loading stats from JSON files.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
Test the time of loading stats of several tables.
Before the PR:
batch1
After the PR:
batch10
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: statistics/handle/handle.go [hunk 1]
```diff
@@ -1107,6 +1107,83 @@ func (h *Handle) StatsMetaCountAndModifyCount(tableID int64) (int64, int64, erro
 	return count, modifyCount, nil
 }
 
+func saveTopNToStorage(ctx context.Context, exec sqlexec.SQLExecutor, tableID int64, isIndex int, histID int64, topN *statistics.TopN) error {
```
Comment (hawkingrei) [inline statistics/handle/handle.go:1110]: when you enable ```bce``` to build it, you can see the ```bce``` case.

```
go build -gcflags="-d=ssa/check_bce/debug=1" ./statistics/handle/...
```

``` 
statistics/handle/handle.go:1122:92: Found IsInBounds
statistics/handle/handle.go:1122:114: Found IsInBounds
statistics/handle/handle.go:1152:23: Found IsInBounds
statistics/handle/handle.go:1169:118: Found IsInBounds
statistics/handle/handle.go:1169:186: Found IsInBounds

```

Comment (hawkingrei) [reply statistics/handle/handle.go:1110]: Can you change the code to avoid some unnecessary bounds checks?

**Review by wjhuang2016 — APPROVED** [review]

File: statistics/handle/handle.go (hunk 1 above)
Comment (xuyifangreeneyes) [reply statistics/handle/handle.go:1110]: OK, I will fix it.

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**